traits/trait_converters.py

```diff
@@ -109,17 +109,43 @@ def trait_for(trait):
         return Trait(trait)
 
 
-def mapped_trait_for(trait):
-    """ Returns the 'mapped trait' definition for a mapped trait.
+def _mapped_trait_default(trait, name, undefined_default, instance):
+    value = getattr(instance, name, undefined_default)
+    if value is undefined_default:
+        return undefined_default
+    return trait.handler.mapped_value(value)
+
+
+def mapped_trait_for(trait, name):
+    """ Returns the 'mapped trait' definition for a mapped trait, the default
+        value of which is a callable that maps the value of the original trait.
+
+        Parameters
+        ----------
+        trait : ctrait.CTrait
+            A trait for which the 'mapped trait' definition is being created.
+        name : str
+            The name of the trait for which the 'mapped trait' definition is
+            being created.
+
+        Returns
+        -------
+        trait_types.Any
+            A definition of the 'mapped trait'
     """
-    from .trait_types import Any
+    from functools import partial
```

Comment:
I think we can lift this import to module-level; no need to make it local to the function.